azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -248,8 +250,20 @@ public void run() throws Exception {
             String.format("Not permitted to proxy as '%s' through Azkaban", effectiveUser)
         );
       }
+
+      final boolean addGroupAndUserForEffectiveUser =
+          this.getSysProps().getBoolean(AZKABAN_ADD_GROUP_AND_USER_FOR_EFFECTIVE_USER, false);
+      if (addGroupAndUserForEffectiveUser) {
+        // If linux group and user needs to be added for effectiveUser before job process starts
+        // then set it up before effective user is used to set permission for any file.
+        info("Adding group and user for effective user: " + effectiveUser);
+        final ExecuteAsUser executeAsUser = new ExecuteAsUser(
+            this.getSysProps().getString(AZKABAN_SERVER_NATIVE_LIB_FOLDER));
+        ExecuteAsUserUtils.addGroupAndUserForEffectiveUser(executeAsUser, effectiveUser);
+      }
+
       // Set parent directory permissions to <uid>:azkaban so user can write in their execution directory
-      // if the directory is not permissioned correctly already (should happen once per execution)
+      // if the directory is not permission correctly already (should happen once per execution)
```

Comment:
Nit: replace with "if the directory does not have correct permissions already"...

Reply:
Good point. Covered.